middlewares/authCondition.js

```diff
@@ -0,0 +1,10 @@
+const authCondition = (authenticate) => {
```

Comment:
Can we name this some thing more appropriate?

Reply:
checking

Reply:
check 2

Reply:
> If I write authProfile it will be too specific. But again I am making very specific middleware.(outlier) @samarpan1738 can you help?

How about `authenticateProfile`?